Vagrant: missing proj6

@miurahr @rouault 

## Expected behavior and actual behavior.

Clean vagrant dev environment provisioning:
```
$ vagrant up
...
==> default: Running provisioner: shell...
    default: Running: /var/folders/4z/cqv7pdjj0hvfj3c15bwy9g4r0000gn/T/vagrant-shell20190208-56589-89bd8w.sh
    default: + NUMTHREADS=2
    default: + [[ -f /sys/devices/system/cpu/online ]]
    default: ++ cut -f 2 -d - /sys/devices/system/cpu/online
    default: + NUMTHREADS=3
    default: + export NUMTHREADS
    default: + rsync -a /vagrant/gdal/ /home/vagrant/gnumake-build-gcc4.8
    default: + rsync -a --exclude=__pycache__ /vagrant/autotest/ /home/vagrant/gnumake-build-gcc4.8/autotest
    default: + echo rsync -a /vagrant/gdal/ /home/vagrant/gnumake-build-gcc4.8/
    default: + echo rsync -a --exclude=__pycache__ /vagrant/autotest/ /home/vagrant/gnumake-build-gcc4.8/autotest
    default: + chmod +x /home/vagrant/gnumake-build-gcc4.8/resync.sh
    default: + cd /home/vagrant/gnumake-build-gcc4.8
    default: + export CCACHE_CPP2=yes
    default: + CCACHE_CPP2=yes
    default: + ARCH_FLAGS=
    default: + AVX2_AVAIL=1
    default: + grep avx2 /proc/cpuinfo
    default: + [[ 1 == \1 ]]
    default: + ARCH_FLAGS=-mavx2
    default: + echo 'AVX2 available on CPU'
    default: AVX2 available on CPU
    default: + CFLAGS=-mavx2
    default: + CXXFLAGS=-mavx2
    default: + CC='ccache gcc'
    default: + CXX='ccache g++'
    default: + LDFLAGS=-lstdc++
    default: + ./configure --prefix=/usr --without-libtool --enable-debug --with-jpeg12 --with-python --with-poppler --with-podofo --with-spatialite --with-java --with-mdb --with-jvm-lib-add-rpath --with-epsilon --with-gta --with-mysql --with-liblzma --with-webp --with-libkml --with-openjpeg=/usr/local --with-armadillo
...
    default: checking for iconv...
    default: yes
    default: checking for working iconv...
    default: yes
    default: checking for iconv declaration...
    default:          extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
    default: using ICONV_CPP_CONST=""
    default: checking for PROJ >= 6 library...
    default: checking for proj_create_from_wkt in -lproj...
    default: no
    default: checking for internal_proj_create_from_wkt in -lproj...
    default: no
    default: checking for internal_proj_create_from_wkt in -linternalproj...
    default: no
    default: configure: error: PROJ 6 symbols not found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

I guess proj6 needs installing in `gdal/scripts/vagrant/gdal.sh`?

## Operating system

Vagrant (Ubuntu 14.04/64bit)

## GDAL version and provenance

master @ 4ee1c997b4c7cffd7fb1d2703dd7317747167482

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -88,7 +88,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     "liblzma-dev",
     "libgeos-dev",
     "libcurl4-gnutls-dev",
-    "libproj-dev",
+    # "libproj-dev",
     "libxml2-dev",
     "libexpat-dev",
     "libxerces-c-dev",
@@ -151,7 +151,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     "mingw-w64-tools",
     "gdb-mingw-w64-target",
     "libgeos-mingw-w64-dev",
-    "libproj-mingw-w64-dev",
+    # "libproj-mingw-w64-dev",
     "cmake3-curses-gui",
     "gdb",
     "gdbserver",
@@ -162,6 +162,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     "libcharls-dev",
     "libgeotiff-dev",
     "libgeotiff-epsg",
+    "sqlite3",
     "sqlite3-pcre",
     "libpcre3-dev",
     "libspatialite-dev",
@@ -196,8 +197,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 	  pkg_cmd << "apt-get --no-install-recommends install -q -y " + packageList.join(" ") << " ; "
 	  config.vm.provision :shell, :inline => pkg_cmd
     scripts = [
+      "install-proj6.sh",
       "gdal.sh",
       "postgis.sh",
+      "install-proj6-mingw.sh",
       "gdal-mingw.sh"
     ];
     scripts.each { |script| config.vm.provision :shell, :privileged => false, :path => "gdal/scripts/vagrant/" << script }

--- a/gdal/scripts/vagrant/gdal-mingw.sh
+++ b/gdal/scripts/vagrant/gdal-mingw.sh
@@ -34,7 +34,8 @@ ln -sf /usr/x86_64-w64-mingw32/lib/libgeos-3-5-0.dll "$HOME/.wine/drive_c/window
 
 (cd /home/vagrant/gnumake-build-mingw-w64
     CC="ccache x86_64-w64-mingw32-gcc" CXX="ccache x86_64-w64-mingw32-g++" LD=x86_64-w64-mingw32-ld \
-    ./configure --prefix=/usr/x86_64-w64-mingw32  --host=x86_64-w64-mingw32  --with-geos #--with-proj
+    ./configure --prefix=/usr/x86_64-w64-mingw32  --host=x86_64-w64-mingw32  --with-geos \
+     --with-sqlite3=/usr/local/x86_64-w64-mingw32 --with-proj=/usr/local/x86_64-w64-mingw32 \
     ln -sf "$PWD/.libs/libgdal-20.dll" "$HOME/.wine/drive_c/windows"
 
     # Python bindings

--- a/gdal/scripts/vagrant/gdal-mingw.sh
+++ b/gdal/scripts/vagrant/gdal-mingw.sh
@@ -27,7 +27,8 @@ wine64 cmd /c dir
 ln -sf /usr/lib/gcc/x86_64-w64-mingw32/4.8/libstdc++-6.dll  "$HOME/.wine/drive_c/windows"
 ln -sf /usr/lib/gcc/x86_64-w64-mingw32/4.8/libgcc_s_sjlj-1.dll  "$HOME/.wine/drive_c/windows"
 ln -sf /usr/x86_64-w64-mingw32/lib/libwinpthread-1.dll  "$HOME/.wine/drive_c/windows"
-ln -sf /usr/x86_64-w64-mingw32/lib/libproj-9.dll "$HOME/.wine/drive_c/windows"
+ln -sf /usr/local/x86_64-w64-mingw32/bin/libsqlite3-0.dll "$HOME/.wine/drive_c/windows"
+ln -sf /usr/local/x86_64-w64-mingw32/bin/libproj-13.dll "$HOME/.wine/drive_c/windows"
 ln -sf /usr/x86_64-w64-mingw32/lib/libgeos_c-1.dll "$HOME/.wine/drive_c/windows"
 ln -sf /usr/x86_64-w64-mingw32/lib/libgeos-3-5-0.dll "$HOME/.wine/drive_c/windows"
 

--- a/gdal/scripts/vagrant/install-proj6-mingw.sh
+++ b/gdal/scripts/vagrant/install-proj6-mingw.sh
@@ -1,0 +1,17 @@
+#!/bin/bash
+
+# abort install if any errors occur and enable tracing
+set -o errexit
+
+SQLITE_YEAR=2019
+SQLITE_VER=3270100
+
+echo "Download, build and install sqlite3 for mingw"
+wget -nc -c -P /var/cache/wget https://sqlite.org/${SQLITE_YEAR}/sqlite-autoconf-${SQLITE_VER}.tar.gz
+tar xf /var/cache/wget/sqlite-autoconf-${SQLITE_VER}.tar.gz
+mkdir sqlite-autoconf-${SQLITE_VER}/build-mingw-w64
+(cd sqlite-autoconf-${SQLITE_VER}/build-mingw-w64; ../configure CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ --build=x86_64-w64-mingw32 --prefix=/usr/local/x86_64-w64-mingw32/ --with-sysroot=/usr/x86_64-w64-mingw32/ && make -j3 && sudo make install )
+
+echo "Build and install proj6 for mingw"
+mkdir proj/build-mingw-w64
+(cd proj/build-mingw-w64; ../configure SQLITE3_CFLAGS="-I/usr/local/x86_64-w64-mingw32/include" SQLITE3_LIBS="-L/usr/local/x86_64-w64-mingw32/lib -lsqlite3"  CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ --host=x86_64-w64-mingw32 --prefix=/usr/local/x86_64-w64-mingw32/ --with-sysroot=/usr/x86_64-w64-mingw32/ && make -j3 && sudo make install )

--- a/gdal/scripts/vagrant/install-proj6.sh
+++ b/gdal/scripts/vagrant/install-proj6.sh
@@ -9,4 +9,4 @@ git clone --depth 1 https://github.com/osgeo/proj.4.git proj
 echo "Build and install proj6..."
 mkdir proj/build-x86_64
 (cd proj; ./autogen.sh)
-(cd proj/build-x86_64; ../configure --prefix=/usr/local/ && make -j3 && sudo make install )
+(cd proj/build-x86_64; CFLAGS='-DPROJ_RENAME_SYMBOLS' CXXFLAGS='-DPROJ_RENAME_SYMBOLS' ../configure --prefix=/usr/local/ && make -j3 && sudo make install && sudo mv /usr/local/lib/libproj.so.13.1.1 /usr/local/lib/libinternalproj.so.13.1.1 && sudo rm /usr/local/lib/libproj.so* && sudo rm /usr/local/lib/libproj.a && sudo rm /usr/local/lib/libproj.la && sudo ln -s libinternalproj.so.13.1.1 /usr/local/lib/libinternalproj.so.13 && sudo ln -s libinternalproj.so.13.1.1 /usr/local/lib/libinternalproj.so)

--- a/gdal/scripts/vagrant/install-proj6.sh
+++ b/gdal/scripts/vagrant/install-proj6.sh
@@ -1,0 +1,12 @@
+#!/bin/bash
+
+# abort install if any errors occur and enable tracing
+set -o errexit
+
+echo "Cloning proj sources..."
+git clone --depth 1 https://github.com/osgeo/proj.4.git proj
+
+echo "Build and install proj6..."
+mkdir proj/build-x86_64
+(cd proj; ./autogen.sh)
+(cd proj/build-x86_64; ../configure --prefix=/usr/local/ && make -j3 && sudo make install )

